### ISSUE TITLE
[Question] training NEP potential with shifted total energies

### ISSUE THREAD

**Issue by JaafarMehrez** [brucefan1983/GPUMD#1637]
Dear GPUMD developers,
I am training a NEP4 potential for the Ti-Si system, targeting TiSi2-C54 (the low-resistivity silicide phase). I have some concern about the training behavior and would appreciate guidance on what I might be doing wrong.

## System and DFT setup
- System: TiSi2-C54, 2 atom types (Si, Ti)
- DFT code: PWmat (PBE functional, SG15 pseudopotentials)
- Dataset: 500 perturbed C54 snapshots with lattice strain, 192 atoms each, generated using the NEPkit
- Train/test split: 80/20 (400 train / 100 test structures)
- The generated dataset can be found in this [repository](https://github.com/JaafarMehrez/SharedFiles/tree/main/037/NEP-train)

For energies, I use an SVD-based least-squares decomposition (shift_energy_to_zero.py, credit to Nan Xu at ZJU) to shift total energies toward zero by subtracting per-atomic reference energies.

basically I have used the default values for nep.in as a starting point: 
```

type          2 Si Ti
generation    200000
```

The attached figures show the loss curves and parity plots of the training

<img width="1770" height="1769" alt="Image" src="https://github.com/user-attachments/assets/68591555-1872-4ff3-b638-b6a91d178fb4" />
<img width="3000" height="3000" alt="Image" src="https://github.com/user-attachments/assets/ef49b4a0-8e71-47b2-8328-3b1b791393e4" />

I find the behaviour of the energy loss strange, similarly for the parity plot of the energy, I am not sure what is the reason for that, any suggsetion or indication is appreciated.

**Comment by brucefan1983** [comment]
The energy shifting for your system should be very simple, as you have a constant number of atoms and  composition for each structurs; it is basically a single value to be added to the energy. 

I noticed that you also have stress values in train.xyz. By default NEP will also use stress to train. Could you add the virial loss to the loss plot and also show the virial/stress parity plots?

**Comment by JaafarMehrez** [comment]
@brucefan1983 Thanks for the quick reply, I have included the virial loss to the loss plot and also included virial/stress parity plots as in the attached figures. 

<img width="1770" height="1769" alt="Image" src="https://github.com/user-attachments/assets/f08b12f3-fd54-4b56-af7d-109a1c5ef175" />
<img width="3000" height="4800" alt="Image" src="https://github.com/user-attachments/assets/0d252bb6-ed92-41fb-a804-72924d52b1f7" />

**Comment by brucefan1983** [comment]
It seems the stress data in train.xyz have a different sign convention than that required by `nep`. 

You can check the manual to confirm:
https://gpumd.org/dev/nep/input_files/train_test_xyz.html

Could you correct the sign a try again?

I will have more suggestions after this is solved.

**Comment by JaafarMehrez** [comment]
Thank you @brucefan1983, I will flip the sign of the stress data and re-train.

**Comment by JaafarMehrez** [comment]
@brucefan1983 following your latest suggestion about the sign convention of stress data, I am getting better training results compared to what I had before, the attached figures shows the training loss and parity plots for the first 100k steps, the training is still on-going. I would like to know if you still have any further suggestions? Thank you   

<img width="1769" height="1769" alt="Image" src="https://github.com/user-attachments/assets/453e0cb6-d21f-4c31-9913-b5f1347d4b05" />

<img width="3000" height="4800" alt="Image" src="https://github.com/user-attachments/assets/5e50150d-4170-4399-9620-12d33f44740f" />

**Comment by brucefan1983** [comment]
The results look good now. Training for 100k steps is sufficient; you don't need more if this is not your final model. My next suggestion is to try a different set of cutoff radii than the default ones (8A, 4 A). Fro Si-Ti system, I guess (6 A, 5 A) is better.

**Comment by JaafarMehrez** [comment]
@brucefan1983 Thank you for the suggestions, I will close as completed.
